pkg/internalapis/featurestates/featurestates.go

```diff
@@ -0,0 +1,418 @@
+							ObjectMeta: metav1.ObjectMeta{Name: SVFeatureStateCRName, Namespace: namespace},
+							Spec: featurestatesv1alpha1.CnsCsiSvFeatureStatesSpec{
+								FeatureStates: pendingCRUpdatesObj.latestFeatureStates,
+							},
+						}
+						err = controllerRuntimeClient.Create(ctx, cnsCsiSvFeatureStates)
+						if err != nil {
+							log.Errorf("failed to create cnsCsiSvFeatureStates CR instance in the "+
+								"namespace: %q, Err: %v", namespace, err)
+							continue
+						}
+						pendingCRUpdatesObj.namespaceUpdateMap[namespace] = false
+						log.Infof("created cnsCsiSvFeatureStates CR instance in the namespace: %q", namespace)
+					} else {
+						log.Errorf("failed to check if cnsCsiSvFeatureStates CR is present in the namespace :%q, err: %v", namespace, err)
+					}
+				}
+			}
+		}()
+	}
+}
+
+// enqueueFeatureStateUpdatesForAllWorkloadNamespaces helps enqueue featurestates updates for all workload namespaces
+func (pendingCRUpdatesObj *pendingCRUpdates) enqueueFeatureStateUpdatesForAllWorkloadNamespaces(ctx context.Context, featurestates []featurestatesv1alpha1.FeatureState) {
+	pendingCRUpdatesObj.lock.Lock()
+	defer pendingCRUpdatesObj.lock.Unlock()
+
+	log := logger.GetLogger(ctx)
+	pendingCRUpdatesObj.latestFeatureStates = featurestates
+	for namespace := range pendingCRUpdatesObj.namespaceUpdateMap {
+		pendingCRUpdatesObj.namespaceUpdateMap[namespace] = true
+		log.Infof("enqueued CR updates for workload namespace: %q", namespace)
+	}
+}
+
+// enqueueFeatureStateUpdatesForWorkloadNamespace enqueues CR updates for specified workload namespaces
+func (pendingCRUpdatesObj *pendingCRUpdates) enqueueFeatureStateUpdatesForWorkloadNamespace(ctx context.Context, namespace string) {
+	pendingCRUpdatesObj.lock.Lock()
+	defer pendingCRUpdatesObj.lock.Unlock()
+
+	log := logger.GetLogger(ctx)
+	pendingCRUpdatesObj.namespaceUpdateMap[namespace] = true
+	log.Infof("enqueued CR updates for workload namespace: %q", namespace)
+}
+
+// configMapAdded is called when configmap is created
+func configMapAdded(obj interface{}) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	ctx = logger.NewContextWithLogger(ctx)
+	log := logger.GetLogger(ctx)
+
+	fssConfigMap, ok := obj.(*v1.ConfigMap)
+	if fssConfigMap == nil || !ok {
+		log.Warnf("configMapAdded: unrecognized object %+v", obj)
+		return
+	}
+	if fssConfigMap.Name == supervisorFeatureStatConfigMapName &&
+		fssConfigMap.Namespace == supervisorFeatureStateConfigMapNamespace {
+		log.Infof("observed fss add: fss: %+v", fssConfigMap.Data)
+		var err error
+		var featureStates []featurestatesv1alpha1.FeatureState
+		for feature, state := range fssConfigMap.Data {
+			var featureState featurestatesv1alpha1.FeatureState
+			featureState.Name = feature
+			featureState.Enabled, err = strconv.ParseBool(state)
+			if err != nil {
+				log.Warnf("failed to parse feature state value: %v for feature: %q", state, feature)
```

Comment:
Why not fail?

Reply:
kept the same logic present in the current fss check implementation based on @shalini-b 's review comments - https://github.com/kubernetes-sigs/vsphere-csi-driver/blob/master/pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator.go#L471-L475

if we fail, then we will need to miss the entire config-map change, even when the single fss value is not correct. 
When parsing fails, setting the feature to `false` would be safe.

Reply:
I think we should fail so that we can catch bugs during dev.

Reply:
ok, in that case until we fix the value of config-map parameter, we can not push the latest fss to namespaces.